libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/signboard/model/MapboxExternalFileResolver.kt

```diff
@@ -0,0 +1,109 @@
+package com.mapbox.navigation.ui.maps.guidance.signboard.model
+
+import android.content.res.AssetManager
+import android.graphics.Typeface
+import com.caverock.androidsvg.SVGExternalFileResolver
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.navigation.utils.internal.LoggerProvider
+
+/**
+ * Mapbox implementation of the [SVGExternalFileResolver] to better support font resolution.
+ *
+ * See [resolveFont].
+ */
+open class MapboxExternalFileResolver(
+    private val assetManager: AssetManager
+) : SVGExternalFileResolver() {
+
+    /**
+     * Called by renderer to resolve font references in &lt;text&gt; elements.
+     *
+     * @param fontFamily Font family as specified in a font-family attribute of css.
+     * @param fontWeight Font weight as specified in a font-weight attribute of css.
+     * @param fontStyle Font style as specified in a font-style attribute of css.
+     *
+     * As per css stylesheet associated as well as the font files supported by the provider
+     * data-set, it supports the following values for font-weight and font-style
+     *
+     * --------------------------------------
+     * font-weight | NORMAL | BOLD
+     * --------------------------------------
+     * font-style | NORMAL | ITALIC | OBLIQUE
+     * --------------------------------------
+     *
+     * Filename should be in the format: $font-family-$font-style$font-weight
+     *
+     * For example for the given input:
+     *
+     * ------------------------------------------------------------------
+     * font-family | font-style | font-weight | fileName
+     * ------------------------------------------------------------------
+     *   SignText  |   Normal   |    Bold     |  SignText-Bold.ttf
+     *   SignText  |   Italic   |   Normal    |  SignText-Italic.ttf
+     *   SignText  |   Italic   |    Bold     |  SignText-ItalicBold.ttf
+     *   SignText  |   Normal   |   Normal    |  SignText.ttf
+     * ------------------------------------------------------------------
+     *
+     * Looking above it's understood that whenever style or weight value is normal it's
+     * represented by empty string
+     */
+    override fun resolveFont(fontFamily: String?, fontWeight: Int, fontStyle: String?): Typeface? {
+
+        try {
+            return getTypeface(fontFamily, fontWeight, fontStyle, ".ttf")
+        } catch (exception: RuntimeException) {
+            LoggerProvider.logger.d(
```

Comment:
```suggestion
            LoggerProvider.logger.e(
```

Reply:
Or maybe also throw an exception, catch it in the processor, and deliver as a failure from the API?

Reply:
If we throw an exception, the entire logic will return an error message and the signboard will not be rendered. Whereas if we don't throw an exception, then the font style may not be applied but the signboard will be rendered. What do you think?
cc @LukasPaczos

Reply:
Yea, typically in other font-related use-cases system would fallback and render with what's available, so let's do the same here. Just make sure to log an error.